### PR TITLE
Autogen native_batch_norm and native_batch_norm_backward

### PR DESCRIPTION
**Pull request by henrytwo**
This PR makes the `native_batch_norm` and `native_batch_norm_backward` ops autogen, and implements their respective shape inference functions.

Previously, these two ops were manually implemented.

cc: @ke1337 @antoniojkim @wconstab @desertfire

### CODE REVIEW

File: test/lazy/test_reuse_ir.py [hunk 1]
```diff
@@ -101,24 +101,5 @@ def testAddSubFallback(self):
...
-
-        device = "lazy"
-        x_lazy = x.detach().clone().to(device=device)
-        bn = bn.to(device=device)
-        for i in range(10):
-            z_lazy = bn(x_lazy)
-            torch._lazy.mark_step()
-
-        torch.testing.assert_close(z.cpu(), z_lazy.cpu())
-        assert metrics.counter_value("IrNodeReused_torch::lazy::TSNativeBatchNormForward") >= 7
```
Comment (antoniojkim) [inline test/lazy/test_reuse_ir.py:124]: Can we maybe not remove this test altogether but rather replace the instance of `TsNativeBatchNormForward` with the codegen requivalent?

Comment (henrytwo) [reply test/lazy/test_reuse_ir.py:124]: That's a good point, I'll try that out

**Comment by henrytwo** [comment]
It looks like all of the CI/CD failures at this point are infrastructure related, so this PR should be ready for review @wconstab @desertfire

File: torch/csrc/lazy/core/shape_inference.cpp [hunk 2]
```diff
@@ -470,6 +470,77 @@ std::vector<Shape> compute_shape_cat(at::TensorList tensors, int64_t dim) {
...
+    double eps) {
+  std::vector<torch::lazy::Shape> shapes;
+  shapes.reserve(3);
+  shapes.emplace_back(input.scalar_type(), input.sizes().vec());
+
+  // A separate mean and var needs to be kept for each channel.
+  TORCH_CHECK(
+      input.sizes().size() >= 2,
+      "Input tensor must have at least batch and channel dimensions!");
+  int64_t num_features = input.sizes().vec()[1];
```
Comment (desertfire) [inline torch/csrc/lazy/core/shape_inference.cpp:490]: Shall we call `input.size(1)`?

Comment (henrytwo) [reply torch/csrc/lazy/core/shape_inference.cpp]: This should be fixed now.

**Review by Gamrix — APPROVED** [review]
Looks good to me!

**Review by desertfire — APPROVED** [review]

**Comment by henrytwo** [comment]
Is there anything I need to do on my end to resolve the CI issues, or is this something that will get fixed later?

**Comment by desertfire** [comment]
> Is there anything I need to do on my end to resolve the CI issues, or is this something that will get fixed later?

Can you rebase your PR to the tip and give it another try?

**Comment by desertfire** [comment]
https://github.com/pytorch/pytorch/runs/6989482831?check_suite_focus=true looks like a true failure.

**Comment by henrytwo** [comment]
> https://github.com/pytorch/pytorch/runs/6989482831?check_suite_focus=true looks like a true failure.

This error should be addressed in this PR: https://github.com/pytorch/pytorch/pull/80060

**Pull request closed without merging**